File: RetrofitTest/app/src/main/java/contract/SecondContract.java
package contract;

/**
 * description:
 * Creat by shiqiang on 2018/5/3 0003 10:37
 */

public interface SecondContract {
    interface Model <T>{

        void loginNet(String userName , String pwd , T t);
    }

    interface View {

        String getUserName();
        String getPassWord();
        void loginSuccess(String success);
        void error(String error);

    }

    interface Presenter {
        //登录联网
        void loginNet();
    }
}

File: RetrofitTest/app/src/main/java/contract/SecondContract.kt
package contract

import info.LoginNetInfo

/**
 * description:
 * Creat by shiqiang on 2018/5/3 0003 10:37
 */

interface SecondContract {
    interface Model{

        fun loginNet(userName: String, pwd: String, t: LoginNetInfo<String>)
    }

    interface View {

        fun getUserName() : String
        fun getPassWord() : String
        fun loginSuccess(success: String)
        fun error(error: String)

    }

    interface Presenter {
        //登录联网
        fun loginNet()
    }
}
